### PR TITLE
Improve code coverage by adding more tests

### PR DESCRIPTION
**Pull request by pkhalaj**
This PR is mainly about the code coverage as mentioned in the title. 

However, as a result of adding more tests, we might need to add some new testing utilities to the [test_utils](https://github.com/pytroll/pytroll-db/tree/master/trolldb/test_utils) package and perform some minor refactoring.

### CODE REVIEW

File: trolldb/config/config.py [hunk 1]
```diff
@@ -46,18 +46,13 @@ def id_must_be_valid(id_like_string: str) -> ObjectId:
     try:
         return ObjectId(id_like_string)
     except InvalidId as e:
-        raise ValueError from e
+        raise ValueError(f"{id_like_string} is not a valid value for an ID.") from e
```
Comment (mraspaud) [inline trolldb/config/config.py:49]: Why change the type of error?

File: trolldb/tests/conftest.py [hunk 2]
```diff
@@ -3,33 +3,66 @@
...
+
+@pytest.fixture()
+def check_log(caplog) -> Callable:
+    """A fixture to check the logs. It relies on the ``caplog`` fixture.
+
+    Returns:
+        A function which can be called to check the log level and message.
+    """
+
+    def _aux(level: str, message: str) -> bool:
```
Comment (mraspaud) [inline trolldb/tests/conftest.py:61]: how about renaming this to something more meaningful? example: `check_log_message_at_level`

File: trolldb/config/config.py (hunk 1 above)
Comment (pkhalaj) [reply trolldb/config/config.py:49]: I have added a note on this to the validator's docstring.

Comment (mraspaud) [reply trolldb/config/config.py:59]: ok, all good then.

**Review by mraspaud — APPROVED** [review]
LGTM

**Pull request merged**